DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateApprovalsController.cs

```diff
@@ -27,8 +27,26 @@ public IActionResult Index()
                 .GetUnapprovedDelegatesWithCustomPromptAnswersForCentre(centreId)
                 .Select(d => new UnapprovedDelegate(d.delegateUser, d.prompts));
 
-            var model = new DelegateApprovalsViewModel(delegates);
+            var model = new DelegateApprovalsViewModel(centreId, delegates);
             return View(model);
         }
+
+        [HttpPost]
+        [Route("/TrackingSystem/Delegates/Approve")]
+        public IActionResult ApproveDelegate(int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            delegateApprovalsService.ApproveDelegate(delegateId, centreId);
+            return RedirectToAction("Index", "DelegateApprovals");
+        }
+
+        [HttpPost]
+        [Route("/TrackingSystem/Delegates/Approve/ApproveForCentre")]

```

Comment:
Could this route be `../Delegates/Approve/All` as everything on the whole site happens per centre (so we don't need for centre everywhere)